### PR TITLE
docs: add info about formats, docker, etc.

### PR DESCRIPTION
**Pull request by szymonlopaciuk**
Signed-off-by: Szymon Łopaciuk <szymon@lopaciuk.eu>

## Checklist:

- [ ] I have all the information that I need (if not, move to `RFC` and look for it).
- [ ] I linked the related issue(s) in the corresponding commit logs.
- [ ] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [ ] My code follows the code style of this project.
- [ ] I've added any new docs if API/utils methods were added.
- [ ] I have updated the existing documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: INSTALL.rst [hunk 1]
```diff
@@ -55,9 +55,79 @@ Finally run the tests to make sure all is setup correctly:
     python setup.py test
```
Comment (david-caro) [inline INSTALL.rst]: This does not work (mixes unit and functional tests)

File: INSTALL.rst [hunk 2]
```diff
@@ -55,9 +55,79 @@ Finally run the tests to make sure all is setup correctly:
...
+
+.. code-block:: console
+
+    docker-compose -f docker-compose.test.yml run --rm unit
+
+
+Installation for testing with inspire-next in Docker
+----------------------------------------------------
+
+Follow the directions to create the virtual environment as above and grab the code from github.
```
Comment (david-caro) [inline INSTALL.rst]: Maybe link to the option? Is the virtualenv really needed? (the section right above says that's not)

File: INSTALL.rst [hunk 3]
```diff
@@ -67,6 +137,8 @@ spider:
     cdvirtualenv src/hepcrawl
     scrapy crawl arXiv -a source_file=file://`pwd`/tests/unit/responses/arxiv/sample_arxiv_record.xml
```
Comment (david-caro) [inline INSTALL.rst]: This fails with:
```
dcaro@pcrcssis001$ scrapy crawl arXiv -a source_file=file://`pwd`/tests/unit/responses/arxiv/sample_arxiv_record.xml
/home/dcaro/Work/repos/per_user/inspirehep/hepcrawl/hepcrawl/spiders/cds_spider.py:21: ScrapyDeprecationWarning: Module `scrapy.spider` is deprecated, use `scrapy.spiders` instead
  from scrapy.spider import XMLFeedSpider
2018-08-24 13:04:54 [scrapy.utils.log] INFO: Scrapy 1.5.1 started (bot: hepcrawl)
2018-08-24 13:04:54 [scrapy.utils.log] INFO: Versions: lxml 4.2.4.0, libxml2 2.9.8, cssselect 1.0.3, parsel 1.5.0, w3lib 1.19.0, Twisted 18.7.0, Python 2.7.15 (default, May 16 2018, 17:50:09) - [GCC 8.1.1 20180502 (Red Hat 8.1.1-1)], pyOpenSSL 18.0.0 (OpenSSL 1.1.0i  14 Aug 2018), cryptography 2.3.1, Platform Linux-4.17.14-202.fc28.x86_64-x86_64-with-fedora-28-Twenty_Eight
2018-08-24 13:04:54 [scrapy.crawler] INFO: Overridden settings: {'NEWSPIDER_MODULE': 'hepcrawl.spiders', 'DUPEFILTER_CLASS': 'scrapy.dupefilters.BaseDupeFilter', 'SPIDER_MODULES': ['hepcrawl.spiders'], 'BOT_NAME': 'hepcrawl', 'USER_AGENT': 'hepcrawl (+http://www.inspirehep.net)', 'EDITOR': 'vim'}
2018-08-24 13:04:55 [scrapy.middleware] INFO: Enabled extensions:
['scrapy.extensions.memusage.MemoryUsage',
 'scrapy.extensions.logstats.LogStats',
 'scrapy.extensions.telnet.TelnetConsole',
 'scrapy.extensions.corestats.CoreStats',
 'hepcrawl.extensions.ErrorHandler']
Unhandled error in Deferred:
2018-08-24 13:04:55 [twisted] CRITICAL: Unhandled error in Deferred:

2018-08-24 13:04:55 [twisted] CRITICAL: 
Traceback (most recent call last):
  File "/home/dcaro/.virtualenvs/hepcrawl/lib/python2.7/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
  File "/home/dcaro/.virtualenvs/hepcrawl/lib/python2.7/site-packages/scrapy/crawler.py", line 98, in crawl
    six.reraise(*exc_info)
  File "/home/dcaro/.virtualenvs/hepcrawl/lib/python2.7/site-packages/scrapy/crawler.py", line 79, in crawl
    self.spider = self._create_spider(*args, **kwargs)
  File "/home/dcaro/.virtualenvs/hepcrawl/lib/python2.7/site-packages/scrapy/crawler.py", line 102, in _create_spider
    return self.spidercls.from_crawler(self, *args, **kwargs)
  File "/home/dcaro/.virtualenvs/hepcrawl/lib/python2.7/site-packages/scrapy/spiders/__init__.py", line 51, in from_crawler
    spider = cls(*args, **kwargs)
  File "/home/dcaro/Work/repos/per_user/inspirehep/hepcrawl/hepcrawl/utils.py", line 378, in wrapper
    ', '.join(disallowed_kwargs),
TypeError: Only underscored kwargs or settings, crawler_settings allowed in <function __init__ at 0x7f2b66503b18>. Check source_file for typos.
```

File: INSTALL.rst [hunk 4]
```diff
@@ -77,7 +149,19 @@ article workflow.
 .. code-block:: console
```
Comment (david-caro) [inline INSTALL.rst]: I would add here a comment about how to run it in docker too.

Comment (david-caro) [reply INSTALL.rst]: xd, it's a few lines after :clown_face:

File: INSTALL.rst [hunk 5]
```diff
@@ -55,9 +55,79 @@ Finally run the tests to make sure all is setup correctly:
...
+        service: base
+      # working_dir: /virtualenv/lib/python2.7/site-packages/hepcrawl
+      working_dir: /hepcrawl_code/hepcrawl
+      command: scrapyd-deploy
+      volumes_from:
+        - static
+      links:
+        - scrapyd
+
+Last step is to deploy the spiders to `scrapyd` (see more in :doc:`Deployment <operations>`):
```
Comment (david-caro) [inline INSTALL.rst]: This operations sections looks outdated, and it's not the inspire-next one, probably you might want to point to https://inspirehep.readthedocs.io/en/latest/harvesting.html, and change the page there (as it's outdated too :/).

File: INSTALL.rst (hunk 4 above)
Comment (szymonlopaciuk) [reply INSTALL.rst]: :+1:

File: docs/inspire_crawler.rst [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+`schedule_crawl <https://github.com/inspirehep/inspire-crawler/blob/master/inspire_crawler/tasks.py>`_
+is used for this purpose. This task schedules a crawler job using the scrapyd API.
+
+Scrapyd then harvests the record.
+
+A harvested record, after going through the pipeline, is passed to ``inspire-crawler``,
+and more precisely a celery task
+`submit_results <https://github.com/inspirehep/inspire-crawler/blob/master/inspire_crawler/tasks.py>`_
+is scheduled. This task creates the workflow for the record that was harvested, and the record,
+appears in the holdingpen.
```
Comment (david-caro) [inline docs/inspire_crawler.rst]: You should probably mention that is the `CrawlResult` the object that is sent with the task, that implements the interaction protocol with inspire-crawler (wrapping the record, any errors and some other metadata).

**Comment by david-caro** [comment]
Just that last nit, after that you can merge :+1:

**Pull request merged**